include/exadg/solvers_and_preconditioners/preconditioners/preconditioner_amg.h

```diff
@@ -474,7 +474,8 @@ class PreconditionerAMG : public PreconditionerBase<Number>
         [&](dealii::LinearAlgebra::distributed::Vector<double> &       dst_double,
             dealii::LinearAlgebra::distributed::Vector<double> const & src_double) {
           preconditioner_ml->vmult(dst_double, src_double);
-        });
+        },
+        true /* copy_dst_entries */);
```

Comment:
See also discussion here https://github.com/dealii/dealii/issues/17978

Reply:
I would say that this interface should not need to copy entries because we call the `ApplyInverse` function of the Epetra operators (I did not look at this code closely yet), we need the initial values only if an iterative solver is called. Is any of the tests failing if we call this particular place with `false` argument as evidence that this function does not overwrite the vector?

Reply:
so the assert we ran into in #715 in debug mode was when we were trying to get the norm of the `dst` vector, see https://github.com/exadg/exadg/issues/715#issuecomment-2566520074
Note that `dst.l2_norm()` was added by me to provoke the assert.
As far as I am concerned, I voted for closing #717 completely while keeping in mind that we are copying data around in some cases. We need to check this next time we do performance measurements.